storm-core/src/jvm/org/apache/storm/ui/UIHelpers.java

```diff
@@ -0,0 +1,319 @@
+                    if (filterParams != null) {
+                        filterHolder.setInitParameters(filterParams);
+                    } else {
+                        filterHolder.setInitParameters(new HashMap<String, String>());
+                    }
+                    context.addFilter(filterHolder, "/*", FilterMapping.ALL);
+                }
+            }
+            context.addFilter(mkAccessLoggingFilterHandle(), "/*", EnumSet.allOf(DispatcherType.class));
+            server.setHandler(context);
+        }
+    }
+
+    public static Map ringResponseFromException(Exception ex) {
+        return ImmutableMap.of(
+                keyword("headers"), new HashMap<>(),
+                keyword("status"), 400,
+                keyword("body"), ex.getMessage()
+        );
+    }
+
+    private static Server removeNonSslConnector(Server server) {
+        for (Connector c : server.getConnectors()) {
+            if (c != null && !(c instanceof SslSocketConnector)) {
+                server.removeConnector(c);
+            }
+        }
+        return server;
+    }
+
+    /**
+     * Construct a Jetty Server instance.
+     */
+    private static Server jettyCreateServer(Integer port, String host, Integer httpsPort) {
+        SelectChannelConnector connector = new SelectChannelConnector();
+        connector.setPort(Utils.getInt(port, 80));
+        connector.setHost(host);
+        connector.setMaxIdleTime(200000);
+
+        Server server = new Server();
+        server.addConnector(connector);
+        server.setSendDateHeader(true);
+
+        if (httpsPort != null && httpsPort > 0) {
+            removeNonSslConnector(server);
+        }
+        return server;
+    }
+
+    /**
+     * Modified version of run-jetty
+     * Assumes configurator sets handler.
+     */
+    public static void stormRunJetty(Integer port, String host, Integer httpsPort, IConfigurator configurator) throws Exception {
+        Server s = jettyCreateServer(port, host, httpsPort);
+        if (configurator != null) {
+            configurator.execute(s);
+        }
+        s.start();
+    }
+
+    public static void stormRunJetty(Integer port, IConfigurator configurator) throws Exception {
+        stormRunJetty(port, null, null, configurator);
+    }
+
+    public static String wrapJsonInCallback(String callback, String response) {
+        return callback + "(" + response + ");";
+    }
+
+    public static Map jsonResponse(Object data, String callback) {
+        return jsonResponse(data, callback, true, null, null);
+    }
+
+    public static Map jsonResponse(Object data, String callback, Long status) {
+        return jsonResponse(data, callback, true, status, null);
+    }
+
+    public static Map jsonResponse(Object data, String callback, Map headers) {
+        return jsonResponse(data, callback, true, null, headers);
+    }
+
+    public static Map jsonResponse(Object data, String callback, boolean needSerialize, Long status, Map headers) {
```

Comment:
Can we please split some of this up?

The final result is ring specific and not really what we want.  Can we have one function for getting the headers, and another function for creating the body.  Then creating the final map can still be in clojure and placed in each of the files that use it.